fineract-investor/src/main/java/org/apache/fineract/investor/service/ExternalAssetOwnerLoanProductAttributesWriteServiceImpl.java

```diff
@@ -0,0 +1,187 @@
+        }
+    }
+
+    private void validateLoanProductAttributeDoesNotExist(Long loanProductId, String attributeKey) {
+        if (externalAssetOwnerLoanProductAttributesRepository.existsByLoanProductIdAndKey(loanProductId, attributeKey)) {
+            throw new ExternalAssetOwnerLoanProductAttributeAlreadyExistsException(
+                    "attributeKey already exists for the loanProductId: " + loanProductId + ". Use PUT call to UPDATE the attribute.");
+        }
+    }
+
+    private void validateLoanProductAttributeKeysMatch(String attributeKeyFromRequest, String attributeKeyFromDB) {
+        if (!attributeKeyFromRequest.equals(attributeKeyFromDB)) {
+            throw new ExternalAssetOwnerLoanProductAttributesException(
+                    "The attribute key of requested update attribute does not match the attribute key from database.");
+        }
+    }
+
+    private void validateExternalAssetOwnerLoanProductAttribute(String attributeKey, String attributeValue) {
+        for (Class<?> implementingClass : implementingClasses) {
+            if (implementingClass.isEnum()) {
+                for (Object obj : implementingClass.getEnumConstants()) {
+                    ExternalAssetOwnerLoanProductAttribute objEnum = (ExternalAssetOwnerLoanProductAttribute) obj;
+                    if (objEnum.getAttributeKey().equals(attributeKey)
+                            && objEnum.getAttributeValue().equals(attributeValue.toUpperCase())) {
+                        return;
+                    }
+                }
+            }
+        }
+        throw new ExternalAssetOwnerLoanProductAttributeInvalidSettlementAttributeException(
+                "The given attribute key or attribute value is not valid.");
+    }
+
+    private ExternalAssetOwnerLoanProductAttributes getLoanProductAttribute(Long attributeId) {
+        Optional<ExternalAssetOwnerLoanProductAttributes> loanProductAttribute = externalAssetOwnerLoanProductAttributesRepository
+                .findById(attributeId);
+        if (loanProductAttribute.isEmpty()) {
+            throw new ExternalAssetOwnerLoanProductAttributeNotFoundException(attributeId);
+        }
+        return loanProductAttribute.get();
+    }
+
+    private ExternalAssetOwnerLoanProductAttributes createExternalAssetOwnerLoanProductAttribute(Long loanProductId, String attributeKey,
+            String attributeValue) {
+        ExternalAssetOwnerLoanProductAttributes attribute = new ExternalAssetOwnerLoanProductAttributes();
+        attribute.setLoanProductId(loanProductId);
+        attribute.setAttributeKey(attributeKey);
+        attribute.setAttributeValue(attributeValue);
+        return attribute;
+    }
+
+    private CommandProcessingResult buildResponseData(ExternalAssetOwnerLoanProductAttributes savedAttribute) {
+        return new CommandProcessingResultBuilder().withResourceIdAsString(savedAttribute.getLoanProductId().toString()).build();
```

Comment:
This seems incorrect: as it will return `resourceIdentifier` in the response, but swagger configuration stated it should return `resourceId`...

Reply:
done.